test/openshift-ci/setup-e2e-tests.sh

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env bash
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is a helper script to run E2E tests on the openshift-ci operator.
+# This script assumes to be run inside a container/machine that has
+# python pre-installed and the `oc` command available. Additional tooling,
+# like kustomize and the mc client are installed by the script if not available.
+# The oc CLI is assumed to be configured with the credentials of the
+# target cluster. The target cluster is assumed to be a clean cluster.
+set -o errexit
+set -o nounset
+set -o pipefail
+
+: "${SKLEARN_IMAGE:=kserve/sklearnserver:latest}"
+: "${KSERVE_CONTROLLER_IMAGE:=quay.io/opendatahub/kserve-controller:latest}"
+: "${KSERVE_AGENT_IMAGE:=quay.io/opendatahub/kserve-agent:latest}"
+: "${KSERVE_ROUTER_IMAGE:=quay.io/opendatahub/kserve-router:latest}"
+: "${STORAGE_INITIALIZER_IMAGE:=quay.io/opendatahub/kserve-storage-initializer:latest}"
+: "${ODH_MODEL_CONTROLLER_IMAGE:=quay.io/opendatahub/odh-model-controller:latest}"
+
+echo "SKLEARN_IMAGE=$SKLEARN_IMAGE"
+echo "KSERVE_CONTROLLER_IMAGE=$KSERVE_CONTROLLER_IMAGE"
+echo "KSERVE_AGENT_IMAGE=$KSERVE_AGENT_IMAGE"
+echo "KSERVE_ROUTER_IMAGE=$KSERVE_ROUTER_IMAGE"
+echo "STORAGE_INITIALIZER_IMAGE=$STORAGE_INITIALIZER_IMAGE"
+
+# Create directory for installing tooling
+# It is assumed that $HOME/.local/bin is in the $PATH
+mkdir -p $HOME/.local/bin
+MY_PATH=$(dirname "$0")
+ODH_MODEL_CTRL_DIR=$MY_PATH/../../
+
+# If Kustomize is not installed, install it
+if ! command -v kustomize &>/dev/null; then
+  echo "Installing Kustomize"
+  curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash -s -- 5.0.1 $HOME/.local/bin
+fi
+
+# If minio CLI is not installed, install it
+if ! command -v mc &>/dev/null; then
+  echo "Installing Minio CLI"
+  curl https://dl.min.io/client/mc/release/linux-amd64/mc --create-dirs -o $HOME/.local/bin/mc
```

Comment:
Please consider downloading these files based on the host's architecture.
Running locally, people on Mac would have trouble.